Source/Core/TaskProcessor.js

```diff
@@ -300,37 +240,27 @@ TaskProcessor.prototype.initWebAssemblyModule = function (webAssemblyOptions) {
     this._worker = createWorker(this);
   }
 
-  const deferred = defer();
-  const processor = this;
   const worker = this._worker;
-  getWebAssemblyLoaderConfig(this, webAssemblyOptions).then(function (
+
+  return getWebAssemblyLoaderConfig(this, webAssemblyOptions).then(function (
     wasmConfig
   ) {
-    return Promise.resolve(canTransferArrayBuffer()).then(function (
-      canTransferArrayBuffer
-    ) {
-      let transferableObjects;
-      const binary = wasmConfig.wasmBinary;
-      if (defined(binary) && canTransferArrayBuffer) {
-        transferableObjects = [binary];
-      }
+    let transferableObjects;
+    const binary = wasmConfig.wasmBinary;
+    if (defined(binary)) {
+      transferableObjects = [binary];
+    }
 
+    const compilePromise = new Promise((resolve) => {
       worker.onmessage = function (event) {
-        worker.onmessage = function (event) {
```

Comment:
Since `scheduleTask` will set the `onmessage`, I don't think this is needed here.